csrc/scheduler/matmul_utils.cpp

```diff
@@ -294,6 +315,98 @@ std::string isMatmulFusionDefinitionSupported(
   return "";
 }
 
+// Assume that tens has a contiguous dimension, and that we will load rows of
+// that dimension, without merging with another dimension first. Then determine
+// the maximum vectorization that can be used.
+//
+// If the argument has no contiguous dimensions, then a vectorization width of 1
+// is returned.
+//
+// These rows can start at any multiple of the non-contiguous strides, so we
+// seek the largest power of 2 that divides all those other dimensions (capped
+// to 16) as well as the data pointer.
+int64_t maxRowVectorization(const at::Tensor& tens) {
+  const int64_t data_ptr_int =
+      static_cast<int64_t>(reinterpret_cast<std::uintptr_t>(tens.data_ptr()));
+  int64_t vec_size = scheduler_utils::maxVectorizationWidth(data_ptr_int);
+  vec_size = std::min(vec_size, (int64_t)(16l));
+  vec_size /= (int64_t)tens.element_size();
+  vec_size = std::max(vec_size, 1l);
+  if (vec_size == 1l) {
+    return vec_size;
+  }
+
+  std::vector<int64_t> strides = tens.strides().vec();
```

Comment:
it will. For an ordinary loop we might vectorize but peel off the last iteration and predicate. However we ar enot doing that for matmul, so we should return 1 in that case correct? If so then we should pass the inner dimension in addition to the non-unit strides to `scheduler_utils::maxVectorizationWidth()`.

Reply:
I added a test that checks that both stride and determine vectorizability.